### ISSUE TITLE
shadow filter + AA + enable assertions = OpenGLException

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#766]
As discussed in the Forum 
+ https://hub.jmonkeyengine.org/t/shadow-filter-anti-aliasing-and-assertions/39601 

combining a shadow filter with anti-aliasing causes an OpenGLException when running with assertions enabled.  I've seen this both with DirectionalLightShadowFilter and with PointLightShadowFilter, but have been unable to reproduce the bug using a DirectionalLightShadowRenderer. The exception only occurs if the filter is enabled.

I discovered the issue in JME version 3.1. It was previously reported in 2015, so it's been around for a while.

Here's the log from a typical failure:
```
Nov 07, 2017 10:08:45 AM com.jme3.system.JmeDesktopSystem initialize
INFO: Running on jMonkeyEngine 3.1-stable
 * Branch: HEAD
 * Git Hash: af04bf9
 * Build Date: 2017-02-13
Nov 07, 2017 10:08:46 AM com.jme3.system.lwjgl.LwjglContext printContextInitInfo
INFO: LWJGL 2.9.3 context running on thread jME3 Main
 * Graphics Adapter: nvd3dumx,nvwgf2umx,nvwgf2umx
 * Driver Version: 21.21.13.7633
 * Scaling Factor: 1
Nov 07, 2017 10:08:46 AM com.jme3.renderer.opengl.GLRenderer loadCapabilitiesCommon
INFO: OpenGL Renderer Information
 * Vendor: NVIDIA Corporation
 * Renderer: GeForce GT 545/PCIe/SSE2
 * OpenGL Version: 4.5.0 NVIDIA 376.33
 * GLSL Version: 4.50 NVIDIA
 * Profile: Compatibility
Nov 07, 2017 10:08:46 AM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio Renderer Information
 * Device: OpenAL Soft
 * Vendor: OpenAL Community
 * Renderer: OpenAL Soft
 * Version: 1.1 ALSOFT 1.15.1
 * Supported channels: 64
 * ALC extensions: ALC_ENUMERATE_ALL_EXT ALC_ENUMERATION_EXT ALC_EXT_CAPTURE ALC_EXT_DEDICATED ALC_EXT_disconnect ALC_EXT_EFX ALC_EXT_thread_local_context ALC_SOFT_loopback
 * AL extensions: AL_EXT_ALAW AL_EXT_DOUBLE AL_EXT_EXPONENT_DISTANCE AL_EXT_FLOAT32 AL_EXT_IMA4 AL_EXT_LINEAR_DISTANCE AL_EXT_MCFORMATS AL_EXT_MULAW AL_EXT_MULAW_MCFORMATS AL_EXT_OFFSET AL_EXT_source_distance_model AL_LOKI_quadriphonic AL_SOFT_buffer_samples AL_SOFT_buffer_sub_data AL_SOFTX_deferred_updates AL_SOFT_direct_channels AL_SOFT_loop_points AL_SOFT_source_latency
Nov 07, 2017 10:08:46 AM com.jme3.audio.openal.ALAudioRenderer initOpenAL
WARNING: Pausing audio device not supported.
Nov 07, 2017 10:08:46 AM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio effect extension version: 1.0
Nov 07, 2017 10:08:46 AM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio max auxiliary sends: 4
numSamples = 2
Nov 07, 2017 10:08:47 AM com.jme3.app.LegacyApplication handleError
SEVERE: An OpenGL error has occured!
org.lwjgl.opengl.OpenGLException: Invalid enum (1280)
	at org.lwjgl.opengl.Util.checkGLError(Util.java:59)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.checkGLError(LwjglAbstractDisplay.java:136)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.runLoop(LwjglAbstractDisplay.java:157)
	at com.jme3.system.lwjgl.LwjglDisplay.runLoop(LwjglDisplay.java:193)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.run(LwjglAbstractDisplay.java:232)
	at java.lang.Thread.run(Thread.java:745)
```

I don't know the root cause. Aside from replacing the filter with a renderer or disabling assertions, an obvious workaround would be to comment out the assertion in LwjglAbstractDisplay.runLoop().

**Comment by stephengold** [comment]
I tried reproducing this issue in a recent master (JME 3.2) build and was unsuccessful, so maybe it doesn't need fixing.

**Comment by stephengold** [comment]
I can confirm that this is fixed in JME 3.2.0-alpha2
